content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx

```diff
@@ -9,40 +10,171 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
+
+The output can be given either as a human-readable summary (the default) or an executable script.
+
+The format of the command is:
+
+```terminal
+npx prisma migrate diff --preview-feature --from-... <source1> --to-... <source2>
+```
+
+where the `--from-...` and `--to-...` options are selected based on the type of database schema source. The supported types of sources are:
 
 - live databases
 - migration histories
 - Prisma data models
 - an empty schema
 
-The format of the command is:
+Both schema sources must use the same database provider. For example, a diff comparing a PostgreSQL data source with a SQLite data source is not supported.
 
-```terminal
-npx prisma migrate diff --preview-feature --from-empty --to-url $PROD_DATABASE
-```
+See also:
 
-For details of the available flags for this command, run:
+- [Migration troubleshooting in production](/guides/database/production-troubleshooting#fixing-failed-migrations-with-migrate-diff-and-db-execute) <span class="guide"></span>
 
-```terminal
-npx prisma migrate diff --help
+#### Prerequisites
+
+Before using the `migrate diff` command, you must define a valid [`datasource`](/concepts/components/prisma-schema/data-sources) within your `schema.prisma` file.
+
+For example, the following `datasource` defines a SQLite database file within the current directory:
+
+```prisma
+datasource db {
+  provider = "sqlite"
+  url      = "file:my-database.db"
+}
 ```
 
-## <inlinecode> prisma db execute </inlinecode>
+#### Options
 
-This command applies a SQL script to the database without interacting with the Prisma migrations table. The script can be passed via a file path or stdin.
+One of the following `--from-...` options is required:
 
-The format of the command is:
+| Options                    | Description                                                          |
+| :------------------------- | :------------------------------------------------------------------- |
+| `--from-url`               | A data source URL                                                    |
+| `--from-migrations`        | Path to the Prisma Migrate migrations directory                      |
+| `--from-schema-datamodel`  | Path to a Prisma schema file, uses the data model for the diff       |
+| `--from-schema-datasource` | Path to a Prisma schema file, uses the `datasource` URl for the diff |
+| `--from-empty`             | Assume that you the data model you are migrating from is empty       |
 
-```terminal
-npx prisma db execute --preview-feature --url $PROD_DATABASE --file script.sql
-```
+One of the following `--to-...` options is required:
 
-For details of the available flags for this command, run:
+| Options                  | Description                                                          |
+| :----------------------- | :------------------------------------------------------------------- |
+| `--to-url`               | A data source URL                                                    |
```

Comment:
Everywhere else we use `datasource` as one word

Reply:
Hm, my understanding from searching the docs is that we use `datasource` when directly referring to the code block, and 'data source' for the generic concept. E.g. [here](https://www.prisma.io/docs/concepts/components/prisma-schema/data-sources): "A data source determines how Prisma connects your database, and is represented by the `datasource`  block in the Prisma schema."

Unless this is actually supposed to be referring to the `datasource` block, in which case it needs backticks around it?

Reply:
Oh I see, everywhere else in this document... yeah, agree, it's a bit muddled. I'll update them all to be consistent.

Reply:
Hopefully that's clearer now